dracut/modules.d/90qubes-pciback/qubes-pciback.sh

```diff
@@ -1,25 +1,53 @@
 #!/bin/sh
 
 type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+unset re HIDE_PCI usb_auth_default usb_in_dom0
 
-# Find all networking devices currenly installed...
-HIDE_PCI="`lspci -mm -n | grep '^[^ ]* "02'|awk '{print $1}'`"
+usb_auth_default=/sys/modules/usbcore/parameters/authorized_default
+usb_in_dom0=false
 
-# ... and optionally all USB controllers...
-if getargbool 0 rd.qubes.hide_all_usb; then
-    HIDE_PCI="$HIDE_PCI `lspci -mm -n | grep '^[^ ]* "0c03'|awk '{print $1}'`"
+modprobe usbcore authorized_default=0
+if getargbool 1 rd.qubes.hide_all_usb; then
+    # Select all networking and USB devices
+    re='0(2|c03)'
+elif grep -Fqe rd.qubes.hide_all_usb /proc/cmdline &&
+    [[ -f "$usb_auth_default" ]] &&
+    echo 0 > "$usb_auth_default"; then
+    # Select only networking devices, but enable USBGuard
+    re='02' usb_in_dom0=true
+else
+    # Cannot disable authorization by default
+    re='0(2|c03)'
+    dfatal 'Cannot securely set up USBGuard - your USB devices in dom0 will not work!'
 fi
 
-HIDE_PCI="$HIDE_PCI `getarg rd.qubes.hide_pci | tr ',' ' '`"
+set -o pipefail
 
+HIDE_PCI=$(set -o pipefail; { lspci -mm -n | awk "/^[^ ]* \"$re/ {print \$1}";}) ||
+    die 'Cannot obtain list of PCI devices to unbind'
+
+manual_pcidevs=$(getarg rd.qubes.hide_pci)
+HIDE_PCI="$HIDE_PCI ${manual_pcidevs//,/ }"
```

Comment:
Can you add also an option to exclude specific device(s) (USB controller) from hiding? The use case would be to hide all the USB controllers except the one for HID, and use USBGuard to ensure it's HID-only. Maybe `rd.qubes.dom0_usb=` ?

Reply:
Done